src/runner/QCheck_base_runner.mli

```diff
@@ -58,10 +58,43 @@ val set_time_between_msg : float -> unit
 (** Set the minimum tiem between messages.
     @since 0.9 *)
 
+
+(** {2 Event handlers} *)
+
+type counter = {
```

Comment:
could this be a `type counter = private {…` to prevent mutation of the fields?

Reply:
That would be a good idea, yes.